libstuff/SHTTPSManager.cpp

```diff
@@ -52,8 +52,9 @@ void SStandaloneHTTPSManager::closeTransaction(Transaction* transaction) {
     // Clean up the socket and done
     _activeTransactionList.remove(transaction);
     _completedTransactionList.remove(transaction);
-    if (transaction->s) {
+    if (transaction->s && !SContains(_closedSocketsList, transaction->s)) {
```

Comment:
This is slow and leaks. You never clear `_closedSocketsList`.

When you close a socket you need to zero it for all the transactions that use it.

Reply:
This is still zero-ing it for all the transactions that use it, right? This is still doing `transaction->s = nullptr;` for all the transactions. I'm just trying to skip closing the socket after it has already been closed and its destructor has been called.

Also, I found a way to check if the socket has been closed already that won't rely on the `_closedSocketsList`. I'll push it. Let me know what you think.

I think I found a decent way to